ohttp/build.rs

```diff
@@ -133,59 +133,111 @@ mod nss {
         assert!(status.success(), "NSS build failed");
     }
 
-    fn dynamic_link() {
-        let libs = if env::consts::OS == "windows" {
-            &["nssutil3.dll", "nss3.dll"]
+    fn nspr_libs() -> Vec<&'static str> {
+        if env::consts::OS == "windows" {
+            vec!["libplds4", "libplc4", "libnspr4"]
         } else {
-            &["nssutil3", "nss3"]
-        };
-        dynamic_link_both(libs);
+            vec!["plds4", "plc4", "nspr4"]
+        }
     }
 
-    fn dynamic_link_both(extra_libs: &[&str]) {
-        let nspr_libs = if env::consts::OS == "windows" {
-            &["libplds4", "libplc4", "libnspr4"]
+    fn dynamic_link() {
+        let mut libs = if env::consts::OS == "windows" {
+            vec!["nssutil3.dll", "nss3.dll"]
         } else {
-            &["plds4", "plc4", "nspr4"]
+            vec!["nssutil3", "nss3"]
         };
-        for lib in nspr_libs.iter().chain(extra_libs) {
+
+        libs.append(&mut nspr_libs());
+
+        for lib in &libs {
             println!("cargo:rustc-link-lib=dylib={}", lib);
         }
     }
 
-    fn static_link() {
+    fn static_softoken_libs(nsslibdir: &Path) -> Vec<&'static str> {
+        let mut static_libs = vec!["pk11wrap_static", "softokn_static", "freebl_static"];
+
+        // NSS optionally builds platform-specific acceleration libraries as
+        // separate static libraries.
+        let accel_libs = &[
+            "gcm-aes-x86_c_lib",
+            "sha-x86_c_lib",
+            "hw-acc-crypto-avx",
+            "hw-acc-crypto-avx2",
+            "armv8_c_lib",
+            "gcm-aes-arm32-neon_c_lib",
+            "gcm-aes-aarch64_c_lib",
+            // NOTE: The intel-gcm-* libraries are already automatically
+            //       included in freebl_static as source files.
+        ];
+
+        // Build rules are complex, so simply check the lib directory to see if
+        // any of the accelerator libraries were built to decide what to
+        // include. Check different variations of the filename to handle
+        // platform differences.
+        for libname in accel_libs {
+            let filename = match env::consts::OS {
+                "windows" => format!("{libname}.lib"),
+                _ => format!("lib{libname}.a"),
+            };
```

Comment:
```suggestion
            let filename = if env::consts::OS == "windows" {
                format!("{libname}.lib")
            } else {
                format!("lib{libname}.a")
            };
```

I have a slight preference for the simpler expression here, even if it is one more line.

Reply:
You and clippy both, it seems. Fixed.